File: twitter-frontend/src/app/model/tweet.ts
import {User} from './user';
import {TweetLike} from './tweetLike';

export class Tweet {
  id: number;
  content: string;
  author: User;
  date: Date;
  likes: TweetLike[];
  sumLikes: number;

  constructor(content: string, author: User, date: Date, likes: TweetLike[]) {
    this.content = content;
    this.author = author;
    this.date = date;
    this.likes = likes;
  }

  setSumLikes(sumLikes: number) {
    this.sumLikes = sumLikes;
  }
}

File: twitter-backend/src/main/java/com/model/Tweet.java
package com.model;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Data
public class Tweet {
    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private Date date = new Date();

    @ManyToOne
    private User author;

    @ElementCollection
    private List<TweetLike> likes;
}